Typescript compilation errors on 5.14.5 with ArrayField

<!-- Please do not submit support requests or "How to" questions here. For that, 
- go to Stack Overflow: https://stackoverflow.com/questions/tagged/react-admin),  or
  go to the react-admin discord server https://discord.gg/GeZF9sqh3N for community support or
- use the Professional Support (https://react-admin-ee.marmelab.com/#support) if you're an Enterprise Edition subscriber. -->

**What you were expecting:**
I'm using an `ArrayField` component:
https://github.com/MTSWebServices/data-rentgen-ui/blob/099f72030c2ba108f05a3312060eddc8b24a2c09/src/components/dataset/DatasetRaList.tsx
```tsx
<List resource="datasets">
    <DatagridConfigurable bulkActionButtons={false}>
        <TextField
            source="data.name"
            label="resources.datasets.fields.name"
            sortable={false}
        />
        <ArrayField
            source="data.tags"
            label="resources.datasets.fields.tags"
        >
            <SomeComponent />
        </ArrayField>
    </DatagridConfigurable>
</List>
```

It compiled properly on 5.14.4, but started to fail after upgrading to 5.14.5

**What happened instead:**
https://github.com/MTSWebServices/data-rentgen-ui/actions/runs/24035103336/job/70092613450?pr=152
```
Error: src/components/dataset/DatasetRaList.tsx(40,21): error TS2322: Type '{ children: Element; source: "data.tags"; label: string; }' is not assignable to type 'IntrinsicAttributes & ArrayFieldBaseProps<{ "data.tags": any; } & { data: any; } & { "data.tags": any; } & { "data.tags": any; } & { "data.tags": any; } & { data: any; }>'.
  Property 'label' does not exist on type 'IntrinsicAttributes & ArrayFieldBaseProps<{ "data.tags": any; } & { data: any; } & { "data.tags": any; } & { "data.tags": any; } & { "data.tags": any; } & { data: any; }>'.
```

This is because #11191 switched from:
```ts
const ArrayFieldImpl(props: ArrayFieldProps) => { ... };

export const ArrayField = genericMemo(ArrayFieldeImpl);
export interface ArrayFieldProps extends FieldProps;
```

to
```ts
const ArrayFieldBaseImpl(props: ArrayFieldBaseProps) => { ... };

export const ArrayFieldBase = genericMemo(ArrayFieldeBaseImpl);
export interface ArrayFieldBaseProps extends FieldBaseProps;

export const ArrayField = ArrayFieldBase;
```

`FieldProps` do have `label` property, but `FieldBaseProps` doesn't.

**Steps to reproduce:**
<!--  Please explain the steps required to duplicate the issue, especially if you are able to provide a sample application. -->

**Related code:**
<!-- If you are able to illustrate the bug or feature request with an example, please provide a sample application via one of the following means: -->

* Preferably, a sandbox forked from 
  - https://codesandbox.io/p/github/marmelab/react-admin-sandbox/main (v5)

  - https://stackblitz.com/github/marmelab/react-admin/tree/master/examples/simple (v5)
  - https://stackblitz.com/github/marmelab/react-admin/tree/4.x/examples/simple (v4)
* A link to a GitHub repo with the minimal codebase to reproduce the issue


```
insert short code snippets here
```

**Other information:**
<!-- List any other information that is relevant to your issue. Stack traces, related issues, suggestions on how to fix, Stack Overflow links, forum links, etc. For visual or layout problems, please include images or animated gifs.-->

**Environment**

* React-admin version: 5.14.5
* Last version that did not exhibit the issue (if applicable): 5.14.4
* React version: 19.2.4
* Browser: NodeJS
* Stack trace (in case of a JS error): see above

--- a/packages/ra-ui-materialui/src/field/ArrayField.stories.tsx
+++ b/packages/ra-ui-materialui/src/field/ArrayField.stories.tsx
@@ -104,6 +104,34 @@ export const Filter = () => (
     </TestMemoryRouter>
 );
 
+export const Label = () => (
+    <TestMemoryRouter>
+        <SimpleShowLayout record={{ id: 123, books }}>
+            <ArrayField source="books" label="List of Books">
+                <SingleFieldList linkType={false}>
+                    <ChipField source="title" />
+                </SingleFieldList>
+            </ArrayField>
+        </SimpleShowLayout>
+    </TestMemoryRouter>
+);
+
+interface Author {
+    id: number;
+    name: string;
+    books: { id: number; title: string }[];
+}
+
+export const Generic = () => (
+    <TestMemoryRouter>
+        <ArrayField<Author> source="books">
+            <SingleFieldList linkType={false}>
+                <ChipField source="title" />
+            </SingleFieldList>
+        </ArrayField>
+    </TestMemoryRouter>
+);
+
 const SortButton = () => {
     const { setSort } = useListContext();
     return (

--- a/packages/ra-ui-materialui/src/field/ArrayField.tsx
+++ b/packages/ra-ui-materialui/src/field/ArrayField.tsx
@@ -1,3 +1,4 @@
+import * as React from 'react';
 import { ArrayFieldBase, type ArrayFieldBaseProps } from 'ra-core';
 
 import type { FieldProps } from './types';
@@ -23,7 +24,11 @@ import type { FieldProps } from './types';
  *
  * @see useListContext
  */
-export const ArrayField = ArrayFieldBase;
+export const ArrayField = <
+    RecordType extends Record<string, any> = Record<string, any>,
+>(
+    props: ArrayFieldProps<RecordType>
+) => <ArrayFieldBase<RecordType> {...props} />;
 
 export interface ArrayFieldProps<
     RecordType extends Record<string, any> = Record<string, any>,

